test/apps/iodemo/io_demo.cc

```diff
@@ -217,13 +307,20 @@ class P2pDemoCommon : public UcxContext {
                                   opts().max_data_size);
     }
 
-    bool send_io_message(UcxConnection *conn, io_op_t op,
-                         uint32_t sn, size_t data_size) {
-        IoMessage *m = _io_msg_pool.get();
+    bool send_io_message(UcxConnection *conn, IoMessage *msg) {
+        VERBOSE_LOG << "sending IO " << io_op_names[msg->op()] << ", sn "
+                    << msg->sn() << " size " << msg->size();
+        return (msg->op() == IO_READ_COMP) ?
+               conn->send_data(msg->buffer(), msg->size(), msg->sn(), msg) :
+               conn->send_io_message(msg->buffer(), msg->size(), msg);
```

Comment:
imo it's confusing that send_io_message eventually can call send_data

Reply:
IMHO it's not more confusing than sending io message by `send_data`. this way all io messages are sent by `send_io_message` but only one type uses lower level "send_data".

Reply:
ok, maybe add comment